src/utils/gauges.ts

```diff
@@ -366,9 +407,345 @@ export async function getGaugeRewardsUserData(
+    await ethCallProvider.tryAll(
+      enumerate(gaugeCount, 0).map((value) =>
+        childGaugeFactoryMultiCall.get_gauge(value),
+      ),
+    )
+  )
+    .map((address) => address && address.toLowerCase())
+    .filter(Boolean) as string[]
+
+  const gaugeMulticallContracts = gaugeAddresses.map((address) =>
+    createMultiCallContract<ChildGauge>(address, CHILD_GAUGE_ABI),
+  )
+
+  const baseGaugeAttributePromises = buildBaseGaugeAttributePromises(
+    gaugeMulticallContracts,
+    ethCallProvider,
+    account,
+  )
+
+  const gaugeRewardCounts = await getGaugeRewardCounts(
+    gaugeMulticallContracts,
+    ethCallProvider,
+  )
+
+  const [
+    gaugeRewardTokens,
+    gaugeBalances,
+    gaugeWorkingBalances,
+    gaugeWorkingSupplies,
+    gaugeTotalSupplies,
+    gaugeLpTokenAddresses,
+    gaugeNames,
+    gaugeKillStatuses,
+  ] = await Promise.all([
+    getGaugeRewardTokens(
+      gaugeMulticallContracts,
+      gaugeRewardCounts,
+      ethCallProvider,
+    ),
+    baseGaugeAttributePromises.gaugeBalancePromise,
+    baseGaugeAttributePromises.gaugeWorkingBalancesPromise,
+    baseGaugeAttributePromises.gaugeWorkingSuppliesPromise,
+    baseGaugeAttributePromises.gaugeTotalSupplyPromise,
+    baseGaugeAttributePromises.gaugeLpTokenAddressesPromise,
+    baseGaugeAttributePromises.gaugeNamesPromise,
+    baseGaugeAttributePromises.gaugeKillStatusesPromise,
+  ])
+
+  const gaugeRewards = await getGaugeRewardsFromTokensSidechain(
+    gaugeRewardCounts,
+    gaugeMulticallContracts,
+    gaugeRewardTokens,
+    ethCallProvider,
+  )
+
+  const sdlRates = await getSDLRatesSidechain(
+    gaugeMulticallContracts,
+    ethCallProvider,
+  )
+  const lpTokenToPool = buildLpTokenToPool(basicPools)
+  const gauges: LPTokenAddressToGauge = buildLpTokenAddressToGauge(
+    chainId,
+    lpTokenToPool,
+    sdlRates,
+    gaugeAddresses,
+    [],
+    [],
+    gaugeRewards,
+    gaugeBalances,
+    gaugeWorkingBalances,
+    gaugeWorkingSupplies,
+    gaugeTotalSupplies,
+    gaugeLpTokenAddresses,
+    gaugeNames,
+    gaugeKillStatuses,
+  )
+
+  return {
+    gaugeCount,
+    gauges,
+  }
+}
+
+async function getSDLRatesSidechain(
+  gaugeMulticallContracts: MulticallContract<ChildGauge>[],
+  ethCallProvider: MulticallProvider,
+): Promise<(BigNumber | null)[]> {
+  const currentTimeStamp = Math.floor(
+    Date.now() / 1000 / BN_DAY_IN_SECONDS.mul(7).toNumber(),
+  )
+  return await ethCallProvider.tryAll(
+    gaugeMulticallContracts.map((contract) =>
+      contract.inflation_rate(Math.floor(currentTimeStamp)),
+    ),
+  )
+}
+
+async function getGaugeRewardsFromTokensSidechain(
+  gaugeRewardCounts: BigNumber[],
+  gaugeMulticallContracts: MulticallContract<ChildGauge>[],
+  gaugeRewardsTokens: (string | null)[][],
+  ethCallProvider: MulticallProvider,
+) {
+  return Promise.all(
+    gaugeRewardCounts.map((count, index) =>
+      ethCallProvider.tryAll(
+        enumerate(count.toNumber(), 0).map((num) => {
```

Comment:
it seems like this function block won't work as expected. Your `map` should pass a `MulticallCall` to `tryAll`, but is instead passing `{...rewardData, token}`. I believe you'll need to wait for the multicall to complete before structuring the resulting data